pymtl3/dsl/Connectable.py

```diff
@@ -449,7 +482,62 @@ def __str__( s ):
   def _str_hook( s ):
     return f"{s._dsl.my_name}"
 
-class NonBlockingCalleeIfc( NonBlockingInterface ):
+class BlockingIfc( Interface ):
+  def construct( s, *args, **kwargs ):
+    raise NotImplementedError("You can only instantiate CallerIfcFL/CalleeIfcFL.")
+
+  def __call__( s, *args, **kwargs ):
+    return s.method( *args, **kwargs )
+
+  def __str__( s ):
+    return s._str_hook()
+
+  def _str_hook( s ):
+    return f"{s._dsl.my_name}"
+
+#-------------------------------------------------------------------------
+# First-class method-based interfaces
+#-------------------------------------------------------------------------
+
+class CalleeIfcRTL( CallIfcRTL ):
+
+  def construct( s, en=None, rdy=None, MsgType=None, RetType=None ):
+    s.MsgType = s.RetType = None
+
+    if en is not None:
+      s.en  = InPort ( Bits1 )
+
+    if rdy is not None:
+      s.rdy = OutPort( Bits1 )
+
+    if MsgType is not None:
+      s.msg = InPort ( MsgType )
+      s.MsgType = MsgType
+
+    if RetType is not None:
+      s.ret = OutPort( RetType )
+      s.RetType = RetType
+
```

Comment:
Can we add some conditions here so that if MsgType is None and RetType is not None then s.msg is the OutPort? This way we don't have to change all of our code that's using send/recv

Reply:
and maybe add a check to make sure at least one port is not None?

Reply:
send/recv is unaffected. Only give/deq is affected